### ISSUE TITLE
Keyboard shortcuts freeze the program for a few seconds if point on self-overlapping path is selected

### ISSUE THREAD

**Issue by hyvyys** [fontforge/fontforge#3827]
Some operations, when triggered with keyboard shortcut, cause the program to freeze for about 5-10 seconds before the operation is performed. These include but probably are not limited to:
 
  * preview toggle
  * coordinates rounding
  * remove overlap

This issue occurs when a point on this path is selected:
![obraz](https://user-images.githubusercontent.com/20814638/61966171-4957e200-afd2-11e9-85ef-0f30494da1dd.png)

At first I thought FF had a problem with redrawing the path because of its nature, but triggering the commands from the menu works smoothly, so that's not it. Also, for preview toggling, sometimes the issue stops occurring even with a node on this path selected after preview is toggled a few times, but reoccurs when a node is moved. Interesting, isn't it?

FontForge 13-April-2019, Windows 10

### Expected behavior
Preview toggled or other operations performed instantly.
### Actual behavior
If a node on some specific paths is selected and the operation is called with its keyboard shortcut, the program freezes for several seconds before performing it, e.g. showing preview/showing back the paths (the issue occurs both ways for preview toggling).

### Steps to reproduce

  1. Open sfd [ampersand.zip](https://github.com/fontforge/fontforge/files/3436628/ampersand.zip).
  2. Open the glyph window on ampersand.
  3. Select any node.
  4. Toggle preview with the keyboard (default is Ctrl+`, if I'm not mistaken – btw a very uncomfortable combination, nasty for the joins of your left thumb, so I've changed that for myself to simple and accessible Z, but I've checked and the issue occurs regardless of the assigned key).

- [x] Screenshot
- [x] The FontForge version and the operating system you're using
- [x] The behavior you expect to see, and the actual behavior
- [x] Steps to reproduce the behavior 
- [ ] \(optional) Possible solution/fix/workaround

**Comment by jtanx** [comment]
The cause of this is pretty simple - on the console, it spews a bunch of

```
SplinePointListIsClockwise: Found error
...
```

Which leads to this backtrace

```
Thread 1 hit Breakpoint 1, SplinePointListIsClockwise (spl=0x51c4698)
    at ../fontforge/splineutil2.c:5052
5052                            fprintf( stderr, "SplinePointListIsClockwise: Found error\n" );
(gdb) bt
#0  SplinePointListIsClockwise (spl=0x51c4698)
    at ../fontforge/splineutil2.c:5052
#1  0x004d4299 in cv_ellistcheck (cv=0x5215fe8, mi=0x52004ac)
    at ../fontforgeexe/charview.c:10915
#2  0x004d4775 in ellistcheck (gw=0x51fe5b0, mi=0x52004ac, UNUSED_e=0x124f790)
    at ../fontforgeexe/charview.c:11003
#3  0x0044af9e in GMenuSearchActionRecursive (gw=0x51fe5b0, mi=0x52003f8,
    action=0x489081e "View.Show.Preview", event=0x124f790, call_moveto=1)
    at ../gdraw/gmenu.c:1127
#4  0x0044b12c in GMenuSearchAction (gw=0x51fe5b0, mi=0x52003f8,
    action=0x489081e "View.Show.Preview", event=0x124f790, call_moveto=1)
    at ../gdraw/gmenu.c:1164
#5  0x0044cfe4 in GMenuBarCheckHotkey (top=0x51fe5b0, g=0x52002f0,
    event=0x124f790) at ../gdraw/gmenu.c:1958
#6  0x0044d22e in GMenuBarCheckKey (top=0x51fe5b0, g=0x52002f0,
    event=0x124f790) at ../gdraw/gmenu.c:2015
#7  0x004151e6 in _GWidget_TopLevel_Key (top=0x51fe5b0, ew=0x51fe5b0,
    event=0x124f790) at ../gdraw/gcontainer.c:507
#8  0x00415a22 in _GWidget_TopLevel_eh (gw=0x51fe5b0, event=0x124f790)
    at ../gdraw/gcontainer.c:682
```

Commenting out the printf here: https://github.com/fontforge/fontforge/blob/master/fontforge/splineutil2.c#L5052

Speeds it up. As for what the *correct* fix is, I'm less sure. Coincidentally, clicking on the `Element` menu will result in the same thing happening; the shortcut is enumerating the menus to find the right action to fire.

**Comment by skef** [comment]
I can reproduce the error messages but the function runs without (perceptual) lag for me on this old Linux box. I'm glancing at this but maybe it's worth thinking about some buffering or whatever under Windows.

**Comment by skef** [comment]
Well, I sort of understand what's going on with the code, but not enough to fix the specific problem at hand. Maybe I'll get back to it after working on my current project some more. 

It appears that FontForge currently has 3 monotonic spline representations. Oof.

**Comment by skef** [comment]
There are intermixed issues here: the slow speed of `stderr` on Windows, the various errors that might be printed, and this particular error, which (because of how `SplinePointListIsClockwise()` is implemented) may be emitted repeatedly for groups of scan lines on a given contour. 

The Remove Overlap messages are likely different, and we should suppress a lot of those. I've opened #4122 for those. The `SplinePointListIsClockwise()` problem was fixed in #4118.
